internal/controller/state/graph/route_common.go

```diff
@@ -873,29 +872,48 @@ func tryToAttachL7RouteToListeners(
 
 // findAttachableListeners returns a list of attachable listeners and whether the listener exists for a non-empty
 // sectionName.
-func findAttachableListeners(sectionName string, listeners []*Listener) ([]*Listener, bool) {
+func findAttachableListeners(ref *ParentRef, listeners []*Listener) ([]*Listener, bool) {
+	sectionName := getSectionName(ref.SectionName)
+
+	// Case 1: sectionName is specified - look for that specific listener
 	if sectionName != "" {
 		for _, l := range listeners {
 			if l.Name == sectionName {
-				if l.Attachable {
+				if l.Attachable && (ref.Port == nil || l.Source.Port == *ref.Port) {
 					return []*Listener{l}, true
 				}
+				if ref.Port != nil && l.Source.Port != *ref.Port {
+					return nil, false
```

Comment:
Wouldn't we want to `continue` in this case? I'm thinking of the situation where multiple listeners have the same section name, and we could eventually match one where the port matches.

Reply:
Oh yeah you're right, for some reason I disassociated them to different Gateways.